### PR TITLE
refact(operator): move operator to latest operator-sdk version

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

This PR moves the jiva operator to the latest operator-sdk version v1.4.2 and fixes issue #43 

As the operator-sdk [upgrade-guide](https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/) has no easy path to move to the latest version from 0.5.2, we took a different approach. We created a new project as the latest operator-sdk only uses controller-runtime and the code structuring is completely new. Then we used the newly generated files in our current project and eliminated all hardcoded operator-sdk dependencies. Now with only controller-runtime and generic k8s packages we can upgrade the k8s version easily without the need for operator-sdk.

### CODE REVIEW

**Review by prateekpandey14 — COMMENTED** [review]
verify the Makefile once , crds gen etc. Also remove any dependency to operator-sdk to gen as well.

File: go.mod [hunk 1]
```diff
@@ -1,67 +1,63 @@
...
-	k8s.io/api v0.17.4
-	k8s.io/apimachinery v0.17.4
-	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/cloud-provider v0.17.4
+	k8s.io/api v0.20.2
+	k8s.io/apimachinery v0.20.2
+	k8s.io/client-go v0.20.2
+	// k8s.io/api v0.20.2
+	// k8s.io/apimachinery v0.20.2
+	// k8s.io/client-go v12.0.0+incompatible
```
Comment (prateekpandey14) [inline go.mod:28]: nit: remove comments

Comment (shubham14bajpai) [reply go.mod]: done

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -58,6 +58,11 @@ jobs:
     - name: verify dependencies
       run: make deps
 
+    - name: verify kubegen
+      run: make verify_kubegen
+      env:
+        GOPATH: /home/runner/work/jiva-operator/go
```
Comment (prateekpandey14) [inline .github/workflows/pull_request.yml:64]: Do we need to  explicitly mention GOPATH here ?

Comment (shubham14bajpai) [reply .github/workflows/pull_request.yml:64]: Yes I tried it without it and it failed. We are doing similar to this in zfs-localpv as well.

File: pkg/controllers/jivavolume_controller.go [hunk 3]
```diff
@@ -893,24 +853,24 @@ func createControllerService(r *ReconcileJivaVolume, cr *jv.JivaVolume) error {
...
-func deleteResource(name, ns string, r *ReconcileJivaVolume, obj runtime.Object) error {
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: ns}, obj)
-	if err != nil && errors.IsNotFound(err) {
-		return nil
-	} else if err != nil {
-		return err
-	}
+// func deleteResource(name, ns string, r *JivaVolumeReconciler, obj runtime.Object) error {
+// 	err := r.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: ns}, obj)
+// 	if err != nil && errors.IsNotFound(err) {
```
Comment (prateekpandey14) [inline pkg/controllers/jivavolume_controller.go:858]: can we remove unused functions as well ?

**Review by prateekpandey14 — APPROVED** [review]
lgtm

**Pull request merged**